pysb/simulator/base.py

```diff
@@ -563,38 +603,54 @@ def __init__(self, simulator, tout, trajectories, squeeze=True,
         yexpr_dtype = zip(expr_names, itertools.repeat(float)) if expr_names \
             else float
 
-        self._yobs = [np.ndarray((len(self.tout[n]),),
-                                 dtype=yobs_dtype) for n in range(self.nsims)]
-        self._yobs_view = [self._yobs[n].view(float).
-                           reshape(len(self._yobs[n]), -1) for n in range(
-            self.nsims)]
-        self._yexpr = [np.ndarray((len(self.tout[n]),),
-                                  dtype=yexpr_dtype) for n in range(
-            self.nsims)]
-        self._yexpr_view = [self._yexpr[n].view(float).reshape(len(
-            self._yexpr[n]), -1) for n in range(self.nsims)]
-        param_values = simulator.param_values
-
-        # loop over simulations
-        sym_names = obs_names + param_names
-        expanded_exprs = [sympy.lambdify(sym_names, expr.expand_expr(),
-                                         "numpy") for expr in exprs]
-        for n in range(self.nsims):
-            simulator._logger.log(EXTENDED_DEBUG, 'Evaluating exprs/obs %d/%d'
-                                  % (n + 1, self.nsims))
-
-            # observables
-            for i, obs in enumerate(model_obs):
-                self._yobs_view[n][:, i] = (
-                    self._y[n][:, obs.species] * obs.coefficients).sum(axis=1)
-
-            # expressions
-            sym_dict = dict((k, self._yobs[n][k]) for k in obs_names)
-            sym_dict.update(dict((p.name, param_values[
-                n // self.n_sims_per_parameter_set][i]) for i, p in
-                            enumerate(self._model.parameters)))
-            for i, expr in enumerate(exprs):
-                self._yexpr_view[n][:, i] = expanded_exprs[i](**sym_dict)
+        if observables_and_expressions is not None and len(
```

Comment:
Just `if observables_and_expressions:` should be OK here. `None` and empty list are both falsey.